File: Assets/UnityChan2D/Demo/FSharpScripts/LoadingController.fs
namespace UnityChan2D.Demo
open System.Collections
open UnityEngine

type LoadingController () =
  inherit MonoBehaviour()

  [<SceneName>]
  [<DefaultValue>] val mutable public nextLevel : string

  member this.Start () = 
    seq {
      yield WaitForSeconds(3.f)
      Application.LoadLevel(this.nextLevel)
    } :?> IEnumerator

File: Assets/UnityChan2D/Demo/Scripts/LoadingController.cs
using UnityEngine;
using System.Collections;
using UniRx;

public class LoadingController : ObservableMonoBehaviour
{
    [SceneName]
    public string nextLevel;

    new public IEnumerator Start()
    {
        yield return new WaitForSeconds(3);

        Application.LoadLevel(nextLevel);
    }
}
